lib/pure/stats.nim

```diff
@@ -261,6 +292,123 @@ proc kurtosisS*[T](x: openArray[T]): float =
   rs.push(x)
   result = rs.kurtosisS()
 
+import random
+
+# Partition using Lomuto partition scheme
+proc partition[T](A: var openArray[T], left, right, K : Natural,
+                                      myCmp : proc(x,y:T):int ): int =
+  assert(left<=K and K<=right)
+  var pInd = K  # Pick `pIndex` as a pivot from the list
+  let pivot = A[pInd]
+  if likely(pInd != right) : swap(A[pInd],A[right])  # Move pivot to end
+  
+  #[
+    elements less than the pivot will be pushed to the left of `pIndex`;
+    elements more than the pivot will be pushed to the right of `pIndex`;
+    equal elements can go either way  ]#
+
+  pInd = left
+  #[ each time we find an element less than or equal to the pivot, `pInd`
+     is incremented, and that element would be placed before the pivot. ]#
+  
+  for i in left ..< right :
+    if myCmp(A[i],pivot) <= 0 :    # A[i] <= pivot
+      if likely(i != pInd) :  swap(A[i],A[pInd])
+      inc(pInd)
+
+  # Move pivot to its place
+  if likely(pInd < right) :  swap(A[pInd],A[right])
+  
+  result= pInd   # A[i] <= pivot  for i in left..pInd 
+
+
+proc quickSelect*[T](A: var openArray[T], left, right, K : Natural, 
+                                          myCmp : proc(x,y:T):int {.nimcall} = cmp[T]): T =
+  ##[
+    Returns the ``K``'th smallest element in a list within `left / right`
+    (i.e., ``left <= K <= right``).
+    
+    If ``K == left`` the smallest element, and if ``K == right`` the largest element of **A** will be returned.
+    
+    >>> **A** will be modified
+    
+  Timings for random arrays on an amd64 / 4GHz machine with  nim c -d:release -d:danger -z 
```

Comment:
Not sure about this - do we want to provide some benchmark results right in the documentation?

Reply:
This mini benchmark should only show that the procedure is very fast. It can be omitted though.
IMHO, these procs to belong to the stats module. The **median** is a robust estimator of the expectation of a random disctribution.

In Python we have  _from statistics import median_,

Reply:
add benchmark tests in `tests/benchmarks/tstats.nim` (and see the readme tests/benchmarks/readme.md) so that people can re-run it